dataset.py

```diff
@@ -5,135 +5,212 @@
-  image_feature_names = ["rh", "temp", "vpd", "atmosphere_oxygen_ratio", "cellulose_oxygen_ratio"]
-  feature_names = ["lat", "lon", "month_of_year"] + image_feature_names
-  rs = RandomState(MT19937(SeedSequence(42)))
-
-  feature_values = {}
-  for name in feature_names:
-    feature_values[name] = []
-
-  for coord in tqdm(sample_site_coordinates):
-    month_start = 0 if monthly else -1
-    month_end = 12 if monthly else 0
-    for month in range(month_start, month_end):
-      samples_collected = 0
-      while samples_collected < samples_per_site:
-        row = {}
-        sample_x, sample_y = 2*(rs.rand(2) - 0.5) * sample_radius
-        sample_x += coord[0]
-        sample_y += coord[1]
-
-        try:
-          for feature, feature_name in zip(features, image_feature_names):
-            row[feature_name] = raster.get_data_at_coords(
-              feature, sample_x, sample_y, month)
-          row["month_of_year"] = month
-          row["lon"] = sample_x
-          row["lat"] = sample_y
-          samples_collected += 1
-
-        except ValueError as e:
-          # masked and out-of-bounds coordinates
-          # note that if sample_radius is zero, we know that looping again
-          # will still result in a ValueError. In this case, we trigger the
-          # loop to finish by incrementing the end condition (samples_collected)
-          # In that case, we end up ignoring the point.
-          # This can also happen with a small positive radius, but one whose
-          # coords return all masked/out of bounds, but this is less likely.
-          if sample_radius == 0:
-            samples_collected += 1
-          continue
-        for key, value in row.items():
-          feature_values[key].append(value)
-
-  samples = pd.DataFrame(feature_values)
-
-  if not monthly:
-    samples.drop("month_of_year", axis=1, inplace=True)
-
-  return samples
-
-def load_sample_data(reference_csv_filename: str) -> pd.DataFrame:
-  """This method loads reference sample data from a CSV, determines the unique
-locations in that CSV and then calls the existing method
-gen_tabular_dataset_with_coords to enrich those locations with features for
-"rh", "temp", "vpd", and  "atmosphere_oxygen_ratio".
-Finally, it joins the CSV provided d18O means values from the CSV with the
-per-loc features"""
+@dataclass
+class DatasetGeographicPartitions:
+    '''
+    Describes the bounds of a geographic area within a dataset.
+    '''
+    min_longitude: float
+    max_longitude: float
+    min_lattitude: float
+    max_lattitude: float
+
+
+_TRAIN_FIXED_BOUNDS = DatasetGeographicPartitions(
+    min_longitude=-62.5,
+    max_longitude=float('inf'),
+    min_lattitude=-5,
+    max_lattitude=float('inf'),
+)
+_VALIDATION_FIXED_BOUNDS = DatasetGeographicPartitions(
+    min_longitude=float('-inf'),
+    max_longitude=-62.5,
+    min_lattitude=-5,
+    max_lattitude=float('inf')
+)
+_TEST_FIXED_BOUNDS = DatasetGeographicPartitions(
```

Comment:
same comment as with the random partition PR.
I would create a PartitionSettings class and have these partition values members of BoundedPartitionSettings and then their use is clearer (as they are arguments on the function).